clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -543,10 +547,14 @@
  */
 public class KafkaConsumer<K, V> implements Consumer<K, V> {
 
+    // Since we're universally extracting timeout durations to milliseconds, this is the maximum duration we can accept.
+    // (2^63)-1 milliseconds is 24.86 days.
+    private static final Duration A_MONTH = Duration.ofMillis(Long.MAX_VALUE);
```

Comment:
24.86 days is not a month :-)

Reply:
Truth! But I wanted to convey the ballpark to help readers and maintainers understand that unpredicated calls will block for quite a long time, but not "effectively forever", which is the mental mistake I made when I read "max long milliseconds". So now, I have this mental scale: "max long" seconds is effectively forever, with milliseconds it's a month-ish, and with nanoseconds it's about a week.